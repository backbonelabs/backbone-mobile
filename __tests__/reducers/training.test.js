import {
  SELECT_LEVEL,
  SELECT_SESSION,
} from '../../app/actions/types';
import trainingReducer from '../../app/reducers/training';

describe('Training Reducer', () => {
  const initialState = {
    plans: [{
      name: '',
      levels: [
        [
          [{
            title: '',
            workout: {},
          }],
        ],
      ],
    }],
    selectedPlanIdx: 0,
    selectedLevelIdx: 0,
    selectedSessionIdx: 0,
  };

  test('should return initial state', () => {
    const trainingState = trainingReducer(initialState, {});
    expect(trainingState).toMatchSnapshot();
    expect(trainingState).toEqual(initialState);
  });

  test('should handle SELECT_LEVEL', () => {
    const newLevel = 1;
    const trainingState = trainingReducer(initialState, {
      type: SELECT_LEVEL,
      payload: newLevel,
    });
    expect(trainingState).toEqual({
      ...initialState,
      selectedLevelIdx: newLevel,
    });
    expect(trainingState).toMatchSnapshot();
  });

  test('should handle SELECT_SESSION', () => {
    const newSession = 2;
    const trainingState = trainingReducer(initialState, {
      type: SELECT_SESSION,
      payload: newSession,
    });
    expect(trainingState).toEqual({
      ...initialState,
      selectedSessionIdx: newSession,
    });
    expect(trainingState).toMatchSnapshot();
  });
});
