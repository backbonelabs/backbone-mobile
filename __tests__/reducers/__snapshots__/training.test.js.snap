// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Training Reducer should handle SELECT_LEVEL 1`] = `
Object {
  "plans": Array [
    Object {
      "levels": Array [
        Array [
          Array [
            Object {
              "title": "",
              "workout": Object {},
            },
          ],
        ],
      ],
      "name": "",
    },
  ],
  "selectedLevelIdx": 1,
  "selectedPlanIdx": 0,
  "selectedSessionIdx": 0,
}
`;

exports[`Training Reducer should handle SELECT_SESSION 1`] = `
Object {
  "plans": Array [
    Object {
      "levels": Array [
        Array [
          Array [
            Object {
              "title": "",
              "workout": Object {},
            },
          ],
        ],
      ],
      "name": "",
    },
  ],
  "selectedLevelIdx": 0,
  "selectedPlanIdx": 0,
  "selectedSessionIdx": 2,
}
`;

exports[`Training Reducer should return initial state 1`] = `
Object {
  "plans": Array [
    Object {
      "levels": Array [
        Array [
          Array [
            Object {
              "title": "",
              "workout": Object {},
            },
          ],
        ],
      ],
      "name": "",
    },
  ],
  "selectedLevelIdx": 0,
  "selectedPlanIdx": 0,
  "selectedSessionIdx": 0,
}
`;
